src/java.base/share/classes/sun/security/x509/AlgorithmId.java

```diff
@@ -169,12 +169,6 @@ public void derEncode (DerOutputStream out) throws IOException {
 
         bytes.putOID(algid);
 
-        // Re-getEncoded() from algParams if it was not initialized
-        if (algParams != null && encodedParams == null) {
-            encodedParams = algParams.getEncoded();
-            // If still not initialized. Let the IOE be thrown.
-        }
-
```

Comment:
This could be a risk change if the caller was not coded like what you do in the EncryptedPrivateKeyInfo.java update.   Did you have a chance to check all caller codes and make sure it is a safe update.

Reply:
Nit on line 168, remove space before "(".